recipes/omniscidb-cuda/recipe/build.sh

```diff
@@ -0,0 +1,143 @@
+
+# Enforce PREFIX instead of BUILD_PREFIX:
+export ZLIB_ROOT=$PREFIX
+export LibArchive_ROOT=$PREFIX
+export Curses_ROOT=$PREFIX
+export Glog_ROOT=$PREFIX
+export Snappy_ROOT=$PREFIX
+export Boost_ROOT=$PREFIX
+export PNG_ROOT=$PREFIX
+export GDAL_ROOT=$PREFIX
+
+# Make sure -fPIC is not in CXXFLAGS (that some conda packages may
+# add):
+export CXXFLAGS="`echo $CXXFLAGS | sed 's/-fPIC//'`"
+
+# Fixes error: no member named 'secure_getenv' in the global namespace
+# Try disabling this for the next omniscidb release, see
+#   https://github.com/omnisci/omniscidb/issues/374
+export CXXFLAGS="$CXXFLAGS -Dsecure_getenv=getenv"
+
+# go overwrites CC and CXX with nonsense (see
+# https://github.com/conda-forge/go-feedstock/issues/47), hence we
+# redefine these below. Reset GO env variables for omniscidb build
+# (IIRC, it is needed for CUDA support):
+#export CGO_ENABLED=1
+#export CGO_LDFLAGS=
+#export CGO_CFLAGS=$CFLAGS
+#export CGO_CPPFLAGS=
+
+# Adjust OPENSSL_ROOT for conda environment. This ensures that
+# openssl is picked up from host environment:
+$INPLACE_SED 's!/usr/local/opt/openssl!\'$PREFIX'!g' CMakeLists.txt
+
+# Avoid picking up boost/regexp header files from local system if
+# there:
+$INPLACE_SED 's!/usr/local!\'$PREFIX'!g' CMakeLists.txt
+
+# Make sure that llvm-config and clang++ are from host environment,
+# otherwise (i) the build environment may contain clang/llvm-4.0.1
+# that will interfer badly with llvmdev/clangdev in the host
+# environment, (ii) UdfTest will fail:
+export PATH=$PREFIX/bin:$PATH
+
+# Linux
+echo "uname=${uname}"
+# must use gcc compiler as llvmdev is built with gcc and there
+# exists ABI incompatibility between llvmdev-7 built with gcc and
+# clang. Using clang with llvmdev-8 is fine.
+COMPILERNAME=clang                      # options: clang, gcc
+
+# Fixes nvcc linker failure (nvcc assumes libraries are in $PREFIX/lib64):
+export LIBRARIES="$LIBRARIES -L$PREFIX/lib"
+
+if [ "$COMPILERNAME" == "clang" ]; then
+    # All these must be picked up from $PREFIX/bin
+    export CC=clang
+    export CXX=clang++
+    export CMAKE_CC=clang
+    export CMAKE_CXX=clang++
+    
+    # Resolves `It appears that you have Arrow < 0.10.0`:
+    export CFLAGS="$CFLAGS -pthread"
+    export LDFLAGS="$LDFLAGS -lpthread -lrt -lresolv"
+else
+    export CC=clang
+    export CXX=  # not used
+    export CMAKE_CC=$HOST-gcc
+    export CMAKE_CXX=$HOST-g++
+fi
+
+##GXX=$HOST-g++         # replace with $GXX
+##GCCVERSION=$(basename $(dirname $($GXX -print-libgcc-file-name)))
+##GXXINCLUDEDIR=$PREFIX/$HOST/include/c++/$GCCVERSION
+# Add gcc include directory to astparser, resolves `not found
+# include file`: cstdint
+# On Ubuntu 18.04 tests pass without this patch, however, the
+# patch is required on Centos and apparently on CI machines
+# (Ubuntu 16.04)
+##$INPLACE_SED 's!arg_vector\[3\] = {arg0, arg1!arg_vector\[4\] = {arg0, arg1, "-extra-arg=-I'$GXXINCLUDEDIR'"!g' QueryEngine/UDFCompiler.cpp
```

Comment:
TODO: arg_vector is consumed by ClangTool. So, one can specify GXXINCLUDEDIR in environment variable `CPLUS_INCLUDE_PATH`

Reply:
done